pkg/tracking/store/sql/trace.go

```diff
@@ -74,3 +80,107 @@ func (s TrackingSQLStore) DeleteTraceTag(
 
 	return nil
 }
+
+func (s TrackingSQLStore) GetTrace(ctx context.Context, reqeustID string) (*entities.TraceInfo, error) {
+	var traceInfo models.TraceInfo
+	if err := s.db.WithContext(
+		ctx,
+	).Where(
+		"request_id = ?", reqeustID,
+	).Preload(
+		"Tags",
+	).Preload(
+		"TraceRequestMetadata",
+	).First(
+		&traceInfo,
+	).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, contract.NewError(
+				protos.ErrorCode_RESOURCE_DOES_NOT_EXIST,
+				fmt.Sprintf(
+					"Trace with request_id '%s' not found.",
+					reqeustID,
+				),
+			)
+		}
+
+		return nil, contract.NewError(
+			protos.ErrorCode_INTERNAL_ERROR, fmt.Sprintf("error getting trace info: %v", err),
+		)
+	}
+
+	return traceInfo.ToEntity(), nil
+}
+
+//nolint:funlen
```

Comment:
Was this really impossible to split up?
Feels like it is possible and you should avoid `nolint` when you can.
We will expect the same from external contributors.